### PR TITLE
chore: Reuses project in tests for `network_container` resource

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Reuses project in tests for `network_container` resource.

The data sources are tested with the test resources so no need to create the same resources twice.

Link to any related issue(s): CLOUDP-237991

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [X] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [X] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/networkcontainer/resource_network_container_test.go [hunk 1]
```diff
@@ -36,16 +34,26 @@ func TestAccNetworkContainerRS_basicAWS(t *testing.T) {
...
+
+					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "provider_name", constant.AWS),
+					resource.TestCheckResourceAttrSet(dataSourceName, "provisioned"),
+
+					resource.TestCheckResourceAttrWith(dataSourcePluralName, "results.#", acc.IntGreatThan(0)),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.id"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.atlas_cidr_block"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provider_name"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provisioned"),
```
Comment (marcosuma) [inline internal/service/networkcontainer/resource_network_container_test.go:52]: is this a trend we're going to use in all resources? testing data sources within the resource test? Or is there something special about this?

Depending on the answer: could we have some comments? (always having in mind readability)

**Review by marcosuma — APPROVED** [review]
LGTM - only one minor comment regarding this new pattern of testing data source into resource

File: internal/service/networkcontainer/resource_network_container_test.go (hunk 1 above)
Comment (lantoli) [reply internal/service/networkcontainer/resource_network_container_test.go:52]: it's not really a new pattern, we have both approaches depending on the resource. We do the same tests to the resource and datasource (checking the same attributes) and we create the same resource to test the datasource as the "resource basic test". 
i'll take it to the tech meeting

File: internal/service/networkcontainer/resource_network_container_test.go [hunk 2]
```diff
@@ -36,16 +34,26 @@ func TestAccNetworkContainerRS_basicAWS(t *testing.T) {
...
 					checkExists(resourceName),
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
 					resource.TestCheckResourceAttr(resourceName, "provider_name", constant.AWS),
 					resource.TestCheckResourceAttrSet(resourceName, "provisioned"),
+
+					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "provider_name", constant.AWS),
+					resource.TestCheckResourceAttrSet(dataSourceName, "provisioned"),
+
+					resource.TestCheckResourceAttrWith(dataSourcePluralName, "results.#", acc.IntGreatThan(0)),
```
Comment (lantoli) [inline internal/service/networkcontainer/resource_network_container_test.go:48]: @marcosuma this follows your idea of doing more generic tests on plural ds (like checking that there is more than 1 as opposed to to a fix number) so they can run in parallel

File: internal/service/networkcontainer/resource_network_container_test.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/networkcontainer/resource_network_container_test.go:52]: A small detail to consider it that when we have acceptance tests that test both the resource and data source you will notice that the test name does not include TestAcc\***RS**\_\*, and instead defines TestAcc*_* ([example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/74687c3088486c3b0c7bb92c72fd03d041a6d6af/internal/service/advancedcluster/resource_advanced_cluster_test.go#L23))

Comment (lantoli) [reply internal/service/networkcontainer/resource_network_container_test.go:52]: thx, good point

**Pull request merged**